### PR TITLE
PBL-52 - Add Pool Volume and TVL to master/detail page

### PR DESCRIPTION
**Pull request by easteregg**
## Issue
https://app.clickup.com/t/865bbh3qj

### CODE REVIEW

File: frontend/apps/pablo/tests/organisms/pool_liquidity_panel.test.tsx [hunk 1]
```diff
@@ -4,7 +4,7 @@ import * as stories from "pablo-storybook/stories/organisms/pool_liquidity_panel
 
 const { Default } = composeStories(stories);
 
-test("renders PoolLiquidityPanel with default args", () => {
+test.skip("renders PoolLiquidityPanel with default args", () => {
```
Comment (andresmechali) [inline frontend/apps/pablo/tests/organisms/pool_liquidity_panel.test.tsx:7]: This is until Subsquid is deployed?

**Review by andresmechali — APPROVED** [review]
LGTM, great job! 🚀

File: frontend/apps/pablo/tests/organisms/pool_liquidity_panel.test.tsx (hunk 1 above)
Comment (easteregg) [reply frontend/apps/pablo/tests/organisms/pool_liquidity_panel.test.tsx:7]: And a bit further than that. needs all data.

File: frontend/apps/pablo/components/Organisms/bonds/PoolShare.tsx [hunk 2]
```diff
@@ -48,24 +51,56 @@ const ItemBox: React.FC<ItemBoxProps> = ({ value, label }) => {
...
+  const [stats, setStats] = useState<GetStatsReturn>(null);
+  useEffect(() => {
+    if (isPoolsLoaded && pool) {
+      getStats(pool).then((result) => {
+        setStats(result);
+      });
+    }
+  }, [isPoolsLoaded, pool]);
+
+  const [assetLeft, assetRight] = input;
```
Comment (vimukthi-git) [inline frontend/apps/pablo/components/Organisms/bonds/PoolShare.tsx:75]: NIT: Left/Right naming might be confusing. You can just use `assetOne` and `assetTwo`

Comment (vimukthi-git) [reply frontend/apps/pablo/components/Organisms/bonds/PoolShare.tsx:75]: Same below for amounts

File: frontend/apps/pablo/components/Organisms/liquidity/AddForm/ConfirmSupplyModal.tsx [hunk 3]
```diff
@@ -60,15 +58,47 @@ export const ConfirmSupplyModal: React.FC<SupplyModalProps & ModalProps> = ({
...
+  }, [isPoolsLoaded, pool]);
 
-  if (poolConfig === null) return null;
-  const assetOne = poolConfig.config.assets[0];
-  const assetTwo = poolConfig.config.assets[1];
+  if (pool === null || stats === null) return null;
+  const assetOne = pool.config.assets[0];
+  const assetTwo = pool.config.assets[1];
+
+  const spotPriceOfATOB = stats[
```
Comment (vimukthi-git) [inline frontend/apps/pablo/components/Organisms/liquidity/AddForm/ConfirmSupplyModal.tsx:78]: NIT: Looks like this logic can be refactored to a function and reused

**Review by vimukthi-git — APPROVED** [review]
LGTM, Note that the derivations of pool share etc., is only valid for 50/50 weight dual asset constant product pools.

File: frontend/apps/pablo/components/Organisms/bonds/PoolShare.tsx (hunk 2 above)
Comment (easteregg) [reply frontend/apps/pablo/components/Organisms/bonds/PoolShare.tsx:75]: Thanks for the review. applied.

File: frontend/apps/pablo/defi/utils/pablo/pools/stats.ts [hunk 4]
```diff
@@ -1,85 +1,154 @@
...
-    if (!data.length) return undefined;
+export function getPoolTVL(stats: GetStatsReturn): BigNumber {
+  if (stats === null) {
+    return new BigNumber(0);
+  }
 
-    let yesterday = data[0].calculatedTimestamp - 1 * DAYS;
-    const yesterdayState = data.find((i) => i.calculatedTimestamp < yesterday);
+  return Object.values(stats).reduce((acc, cur) => {
+    acc = acc.plus(cur.total.volume.multipliedBy(cur.spotPrice));
```
Comment (vimukthi-git) [inline frontend/apps/pablo/defi/utils/pablo/pools/stats.ts:98]: This seems off, why we are taking volume for TVL?

File: frontend/apps/pablo/defi/utils/pablo/pools/stats.ts [hunk 5]
```diff
@@ -1,85 +1,154 @@
...
+export function getPoolVolume(stats: GetStatsReturn): BigNumber {
+  if (stats === null) {
+    return new BigNumber(0);
+  }
 
-        _24HourVolume = data[0].totalVolume.minus(
-            yesterdayState.totalVolume
-        );
+  return Object.values(stats).reduce((acc, cur) => {
+    acc = acc.plus(cur.total.liquidity.multipliedBy(cur.spotPrice));
```
Comment (vimukthi-git) [inline frontend/apps/pablo/defi/utils/pablo/pools/stats.ts:110]: This seems off, why we are taking TVL for volume?

File: frontend/apps/pablo/defi/utils/pablo/pools/stats.ts (hunk 4 above)
Comment (easteregg) [reply frontend/apps/pablo/defi/utils/pablo/pools/stats.ts:98]: ah it should be the other way around right?

File: frontend/apps/pablo/defi/utils/pablo/pools/stats.ts (hunk 5 above)
Comment (easteregg) [reply frontend/apps/pablo/defi/utils/pablo/pools/stats.ts]: Nice catch, I've replaced it.

**Review by vimukthi-git — APPROVED** [review]

**Pull request merged**